// nativescript-feedback/feedback.ios.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const feedback_common_1 = require("./feedback.common");
exports.FeedbackPosition = feedback_common_1.FeedbackPosition;
exports.FeedbackType = feedback_common_1.FeedbackType;
class Feedback extends feedback_common_1.FeedbackCommon {
    show(options) {
        return new Promise((resolve, reject) => {
            let icon = options.icon ? UIImage.imageNamed(options.icon) : null;
            let hideOnSwipe = true;
            let hideOnTap = true;
            let message = ISMessages.cardAlertWithTitleMessageIconImageDurationHideOnSwipeHideOnTapAlertTypeAlertPosition(options.title, options.message, icon, options.duration ? options.duration / 1000 : 4.0, hideOnSwipe, hideOnTap, Feedback.getType(options.type), Feedback.getPosition(options.position));
            if (options.backgroundColor) {
                message.alertViewBackgroundColor = options.backgroundColor.ios;
            }
            if (options.titleColor) {
                message.titleLabelTextColor = options.titleColor.ios;
            }
            const titleSize = options.titleSize || 16;
            const messageSize = options.messageSize || 13;
            if (options.titleFont) {
                message.titleLabelFont = UIFont.fontWithNameSize(options.titleFont, titleSize);
            }
            else {
                message.titleLabelFont = UIFont.boldSystemFontOfSize(titleSize);
            }
            if (options.messageFont) {
                message.messageLabelFont = UIFont.fontWithNameSize(options.messageFont, messageSize);
            }
            else {
                message.messageLabelFont = UIFont.systemFontOfSize(messageSize);
            }
            if (options.messageColor) {
                message.messageLabelTextColor = options.messageColor.ios;
            }
            message.showDidHide(() => {
                if (options.onTap) {
                    options.onTap();
                }
            }, (hidden) => {
            });
            resolve();
        });
    }
    hide(options) {
        return new Promise((resolve, reject) => {
            ISMessages.hideAlertAnimated(true);
            resolve();
        });
    }
    static getType(type) {
        if (type === undefined || type === null || type === feedback_common_1.FeedbackType.Custom) {
            return 4;
        }
        else if (type === feedback_common_1.FeedbackType.Warning) {
            return 2;
        }
        else if (type === feedback_common_1.FeedbackType.Error) {
            return 1;
        }
        else if (type === feedback_common_1.FeedbackType.Info) {
            return 3;
        }
        else {
            return 0;
        }
    }
    static getPosition(position) {
        if (!position || position === feedback_common_1.FeedbackPosition.Top) {
            return 0;
        }
        else {
            return 1;
        }
    }
}
exports.Feedback = Feedback;
